src/components/line-graph.js

```diff
@@ -0,0 +1,115 @@
+import * as d3 from 'd3';
+
+/**
+ * @name LineGraph
+ * @description
+ * @extends HTMLElement
+ * Generic line graph component for reusability
+ */
+export class LineGraph extends HTMLElement {
+ 
+    constructor() {
+        super();
+    }
+
+    /**
+     * @name connectedCallback
+     * @description
+     * Call back for when the component is attached to the DOM
+     */
+    connectedCallback() {
+        this.innerHTML = "<svg></svg>";
+        this.renderGraph(this.parseData(this.dataset.graph));
+    }
+
+    /**
+     * @name disconnectedCallback
+     * @description
+     * Call back for when the component is detached from the DOM
+     */
+    disconnectedCallback() {}
+
+    /**
+     * @name parseData
+     * @param {Object} data 
+     * @description
+     * Parses data into an array while converting time to a proper
+     * Javascript date object
+     */
+    parseData(data) {
+        data = JSON.parse(data);
+        var results = [];
+        for (var i = 0; i < data.length; i++) {
+            var item = {};
+            var key = Object.keys(data[i])[0];
+            item.time = new Date(data[i][key].time);
+            item.value = data[i][key].value;
+            results.push(item);
+        }
+        return results;
+    }
+
+    /**
+     * @name renderGraph
+     * @param {Object} data 
+     * @description
+     * Renders the graph using d3js
+     */
+    renderGraph(data) {
+
+        // Setup the margins and height, width
+        var margin = JSON.parse(this.dataset.margin);
+        var height = this.dataset.height;
+        var width = this.dataset.width;
+
+        // Setup the svg element in the DOM
+        var svg = d3.select('svg')
```

Comment:
Would you need to first narrow this selection to the current HTML element? Otherwise, I thought this by default will select any and all `<svg>` elements in the document.

Reply:
hmm.  Good question/catch.  I think I can probably select the parent element.  Let me give it a try just to be on the safe side.

Reply:
Oh I just remembered it won't be a problem because of the shadow dom that we get from the HTMLElement